processing/src/main/java/org/apache/druid/query/CacheStrategy.java

```diff
@@ -43,10 +43,11 @@ public interface CacheStrategy<T, CacheType, QueryType extends Query<T>>
    * @param query            the query to be cached
    * @param willMergeRunners indicates that {@link QueryRunnerFactory#mergeRunners(QueryProcessingPool, Iterable)} will be
    *                         called on the cached by-segment results
+   * @param bySegment        segment level or result level cache
    *
    * @return true if the query is cacheable, otherwise false.
    */
-  boolean isCacheable(QueryType query, boolean willMergeRunners);
+  boolean isCacheable(QueryType query, boolean willMergeRunners, boolean bySegment);
```

Comment:
we need to retain the original method for backward incompatibility.

Reply:
get this.